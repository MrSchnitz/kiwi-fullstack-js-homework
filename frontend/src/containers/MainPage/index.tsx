import React from "react";
import styled from "styled-components";
import { withNavigation } from "../../hoc/NavBar";
import Phone from "../../components/Phone";

function MainPage() {
  return (
    <MainPageLayout>
      <Phone
        displayText={"Phone"}
        predictionWords={[]}
        predictionWordsLoading={false}
        onDeleteChar={() => null}
        onKeyPressed={() => null}
        onPredictionSelect={() => null}
      />
    </MainPageLayout>
  );
}

const MainPageLayout = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default withNavigation(MainPage);
