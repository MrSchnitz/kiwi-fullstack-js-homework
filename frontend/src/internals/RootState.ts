import { connectRouter } from "connected-react-router";
import { Action, ThunkAction } from "@reduxjs/toolkit";
import { all } from "redux-saga/effects";
import { createBrowserHistory } from "history";

export const history = createBrowserHistory();

export interface RootState {
  // API
  // Containers

  // Router
  router?: any;
}

export type AppThunk = ThunkAction<void, RootState, unknown, Action<string>>;

export const mainReducers = {
  // API
  // Containers

  // Router
  router: connectRouter(history),
};

/**
 * IMPORT EVERY OTHER NOT SOMEWHERE ELSE INJECTED SAGA
 */
export default function* rootSaga() {
  yield all([]);
}
