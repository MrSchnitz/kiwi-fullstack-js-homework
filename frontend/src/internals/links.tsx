import React, { ReactNode } from "react";
import { Airplane, Partners } from "@kiwicom/orbit-components/lib/icons";

export interface LinkInterface {
  title: string;
  url: string;
  icon?: ReactNode;
}

export const Links = {
  mainPage: "/",
  descriptionPage: "/description",
};

export const NavigationLinks: LinkInterface[] = [
  {
    title: "Main",
    url: Links.mainPage,
    icon: <Airplane />,
  },
  {
    title: "Homework Description",
    url: Links.descriptionPage,
    icon: <Partners />,
  },
];
