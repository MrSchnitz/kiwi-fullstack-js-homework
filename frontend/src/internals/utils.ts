export const ENV = process.env.BE || "http://localhost:4000";
