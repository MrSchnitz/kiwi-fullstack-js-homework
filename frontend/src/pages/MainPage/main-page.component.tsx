import React from "react";
import Phone from "../../components/Phone/phone.component";
import {useDispatch, useSelector} from "react-redux";
import {MainAPI, selectNumbers, selectPredictionWords, selectPredictionWordsLoading, selectText,} from "./api/MainAPI";
import {WordType} from "../../models/Word";
import {MainPageLayout} from "./main-page.styles";

function MainPage() {
  const dispatch = useDispatch();

  const numbers: number[] = useSelector(selectNumbers);

  const text: string = useSelector(selectText);

  const predictionWords: WordType[] = useSelector(selectPredictionWords);

  const predictionWordsLoading: boolean = useSelector(
    selectPredictionWordsLoading
  );

  return (
    <MainPageLayout>
      <Phone
        displayText={`${text} ${numbers.toString().replace(/,/g, "")}`}
        predictionWords={predictionWords}
        predictionWordsLoading={predictionWordsLoading}
        onKeyPressed={(key: number) => dispatch(MainAPI.addNumber(key))}
        onDeleteChar={() => dispatch(MainAPI.deleteText())}
        onPredictionSelect={(word: WordType) =>
          dispatch(MainAPI.changeText(word))
        }
      />
    </MainPageLayout>
  );
}

export default MainPage;
