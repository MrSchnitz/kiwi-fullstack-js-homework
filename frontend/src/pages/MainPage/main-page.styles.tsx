import styled from "styled-components";

export const MainPageLayout = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;
