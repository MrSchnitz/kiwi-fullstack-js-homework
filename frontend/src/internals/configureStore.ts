import { routerMiddleware } from "connected-react-router";
import {
  combineReducers,
  configureStore,
  getDefaultMiddleware,
} from "@reduxjs/toolkit";
import rootSaga, { mainReducers } from "./RootState";
import createSagaMiddleware from "redux-saga";
import { history } from "./RootState";

export function configureAppStore() {
  const reduxSagaMonitorOptions = {};
  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with saga middleware and router middleware
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const store = configureStore({
    reducer: combineReducers({ ...mainReducers }),
    middleware: [...getDefaultMiddleware(), ...middlewares],
    devTools: process.env.NODE_ENV !== "production",
  });

  // Add all combined Sagas to middleware
  sagaMiddleware.run(rootSaga);

  return store;
}
