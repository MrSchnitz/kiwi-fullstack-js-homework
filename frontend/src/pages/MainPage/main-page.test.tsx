import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import MainPage from "./main-page.component";
import { configureAppStore } from "../../internals/configureStore";
import { Provider } from "react-redux";
import * as redux from "react-redux";

describe("Main page", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    const store = configureAppStore();
    const spy = jest.spyOn(redux, "useSelector");
    spy.mockReturnValue({ text: "test" });

    wrapper = shallow(
      <Provider store={store}>
        <MainPage />
      </Provider>
    );
  });

  it("should render Main page", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
