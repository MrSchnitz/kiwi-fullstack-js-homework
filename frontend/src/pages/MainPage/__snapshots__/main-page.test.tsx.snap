// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page should render Main page 1`] = `ShallowWrapper {}`;
